src/Agent.Worker/TestResults/JunitResultReader.cs

```diff
@@ -0,0 +1,313 @@
+﻿using Microsoft.TeamFoundation.TestManagement.WebApi;
+using Microsoft.VisualStudio.Services.Agent.Util;
+using Microsoft.VisualStudio.Services.WebApi;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Xml;
+
+namespace Microsoft.VisualStudio.Services.Agent.Worker.TestResults
+{
+    public class JUnitResultReader : AgentService, IResultReader
+    {
+        public Type ExtensionType => typeof(IResultReader);
+        public string Name => "JUnit";
+
+        private IExecutionContext _executionContext;
+        private bool _addResultsFileToRunLevelAttachments = true;
+
+        /// <summary>
+        /// Reads a JUnit results file from disk, converts it into a TestRunData object.        
+        /// </summary>
+        /// <param name="filePath"></param>
+        /// <param name="runContext"></param>
+        /// <returns></returns>
+        public TestRunData ReadResults(IExecutionContext executionContext, string filePath, TestRunContext runContext = null)
+        {
+            _executionContext = executionContext;
```

Comment:
You don't need this private variable. Its not used anywhere else. Keep it as local variables.